lib/endpoints/class-wp-rest-comments-controller.php

```diff
@@ -720,7 +823,7 @@ protected function prepare_item_for_database( $request ) {
 		}
 
 		if ( isset( $request['parent'] ) ) {
-			$prepared_comment['comment_parent'] = $request['parent'];
+			$prepared_comment['comment_parent'] = (int) $request['parent'];
```

Comment:
Not sure why it would be an array but definitely should be type cast to int.  I think maybe putting the new logic into a protected method somewhere in the controller would be nice, to keep `create_permissions_check()` clean. When your patch lands in core we can just cut the lines back out.